### ISSUE TITLE
Verbrauchswerte Last X

### ISSUE THREAD

**Issue by serpentes80** [DrozmotiX/ioBroker.sourceanalytix#460]
Hallo,

bei mir werden die Verbrauchswerte "Last Day" nicht befüllt. 
Statt dessen läuft "Current Day" immer weiter was dann zu falschen Werten bei mir führt.
Irgend ne Idee?

**Comment by DutchmanNL** [comment]
kommt da was im log bei dir nachts zwischen 23.50 und 00.05 ?

**Comment by serpentes80** [comment]
das hier hätte ich im Angebot:

`
<html>
<body>


2021-08-10 00:00:00.011 - info: sourceanalytix.0 (1082) Reset start values for : shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned
--
2021-08-10 00:00:00.022 - info: sourceanalytix.0 (1082) Memory values before reset : null
2021-08-10 00:00:00.023 - error: sourceanalytix.0 (1082) [[resetStartValues] shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned] error: Cannot read property 'calcValues' of null, stack: TypeError: Cannot read property 'calcValues' of null
at CronJob. (/opt/iobroker/node_modules/iobroker.sourceanalytix/main.js:621:54)
2021-08-10 00:00:00.026 - error: sourceanalytix.0 (1082) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-08-10 00:00:00.026 - error: sourceanalytix.0 (1082) unhandled promise rejection: Cannot read property 'captureException' of undefined
2021-08-10 00:00:00.027 - error: sourceanalytix.0 (1082) TypeError: Cannot read property 'captureException' of undefined
at Sourceanalytix.errorHandling (/opt/iobroker/node_modules/iobroker.sourceanalytix/main.js:1647:37)
at CronJob. (/opt/iobroker/node_modules/iobroker.sourceanalytix/main.js:849:12)
2021-08-10 00:00:00.027 - error: sourceanalytix.0 (1082) Cannot read property 'captureException' of undefined
2021-08-10 00:00:00.031 - info: sourceanalytix.0 (1082) SourceAnalytix stopped, now you have to calculate by yourself :'( ...
2021-08-10 00:00:00.032 - info: sourceanalytix.0 (1082) terminating
2021-08-10 00:00:00.035 - warn: sourceanalytix.0 (1082) Terminated (UNCAUGHT_EXCEPTION): Without reason
2021-08-10 00:00:00.060 - info: host.buanet-iobroker1 instance system.adapter.daswetter.0 started with pid 317
2021-08-10 00:00:00.033 - warn: jarvis.0 (975) Read-only state "jarvis.0.info.data" has been written without ack-flag with value "{"event":"time:midnight"}"
2021-08-10 00:00:00.642 - error: host.buanet-iobroker1 instance system.adapter.sourceanalytix.0 terminated with code 6 (UNCAUGHT_EXCEPTION)
2021-08-10 00:00:00.643 - info: host.buanet-iobroker1 Restart adapter system.adapter.sourceanalytix.0 because enabled
2021-08-10 00:00:01.114 - info: daswetter.0 (317) Plugin sentry Sentry Plugin disabled for this process because sending of statistic data is disabled for the system
2021-08-10 00:00:01.127 - info: daswetter.0 (317) starting. Version 3.0.7 in /opt/iobroker/node_modules/iobroker.daswetter, node: v12.22.1, js-controller: 3.3.15
2021-08-10 00:00:02.007 - info: mihome-vacuum.0 (493) State value to set for "mihome-vacuum.0.info.water_box" has to be type "string" but received type "boolean"
2021-08-10 00:00:15.388 - info: daswetter.0 (317) still wrong data structure from server received! repaired...
2021-08-10 00:00:18.853 - info: daswetter.0 (317) Terminated (NO_ERROR): Without reason
2021-08-10 00:00:19.366 - info: daswetter.0 (317) cleaned everything up...
2021-08-10 00:00:19.380 - info: host.buanet-iobroker1 instance system.adapter.daswetter.0 terminated with code 0 (NO_ERROR)
2021-08-10 00:00:23.125 - info: mihome-vacuum.0 (493) State value to set for "mihome-vacuum.0.info.water_box" has to be type "string" but received type "boolean"
2021-08-10 00:00:30.671 - info: host.buanet-iobroker1 instance system.adapter.sourceanalytix.0 started with pid 360
2021-08-10 00:00:31.880 - info: sourceanalytix.0 (360) Plugin sentry Sentry Plugin disabled for this process because sending of statistic data is disabled for the system
2021-08-10 00:00:31.894 - info: sourceanalytix.0 (360) starting. Version 0.4.9 in /opt/iobroker/node_modules/iobroker.sourceanalytix, node: v12.22.1, js-controller: 3.3.15
2021-08-10 00:00:31.917 - info: sourceanalytix.0 (360) Welcome to SourceAnalytix, making things ready ...
2021-08-10 00:00:32.076 - error: sourceanalytix.0 (360) Cannot execute calculations for shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned adjust settings !
2021-08-10 00:00:32.076 - error: sourceanalytix.0 (360) No cost type defined for shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned, please Select Type of calculation at state setting
2021-08-10 00:00:32.094 - info: sourceanalytix.0 (360) Initialising (1 of 4) state shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned
2021-08-10 00:00:32.096 - error: sourceanalytix.0 (360) Cannot handle initialisation for shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned
2021-08-10 00:00:32.096 - info: sourceanalytix.0 (360) Initialization (1 of 4) finished for : shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned
2021-08-10 00:00:32.097 - info: sourceanalytix.0 (360) Initialising (2 of 4) state shelly.0.SHSW-PM#84CCA8AF1BBB#1.Relay0.Power
2021-08-10 00:00:33.895 - info: sourceanalytix.0 (360) Initialization (2 of 4) finished for : shelly.0.SHSW-PM#84CCA8AF1BBB#1.Relay0.Power
2021-08-10 00:00:33.895 - info: sourceanalytix.0 (360) Initialising (3 of 4) state 0_userdata.0.shelly_total.Shelly_3em_total
2021-08-10 00:00:35.424 - info: sourceanalytix.0 (360) Initialization (3 of 4) finished for : 0_userdata.0.shelly_total.Shelly_3em_total
2021-08-10 00:00:35.424 - info: sourceanalytix.0 (360) Initialising (4 of 4) state 0_userdata.0.shelly_total.Verschenkte_kWh
2021-08-10 00:00:36.639 - info: sourceanalytix.0 (360) Initialization (4 of 4) finished for : 0_userdata.0.shelly_total.Verschenkte_kWh
2021-08-10 00:00:36.652 - info: sourceanalytix.0 (360) SourceAnalytix initialisation finalized, will handle calculations ... for 4 states


</body>
</html>`

**Comment by DutchmanNL** [comment]
Sehr interessant vielen dank du hast da einen big entdeckt wo ich bereits Meldung en in Sentry drüber empfangen haben aber nicht verstehen konnte durch es verursacht wird mit deinem log kan ich es endlich reproduzieren !!!

der adapter crasht nachts um 00.00 Uhr wodurch die tageswerte nicht resettet werden : 

```
2021-08-10 00:00:00.022 - info: sourceanalytix.0 (1082) Memory values before reset : null
2021-08-10 00:00:00.023 - error: sourceanalytix.0 (1082) [[resetStartValues] shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned] error: Cannot read property 'calcValues' of null, stack: TypeError: Cannot read property 'calcValues' of null
at CronJob. (/opt/iobroker/node_modules/iobroker.sourceanalytix/main.js:621:54)
2021-08-10 00:00:00.026 - error: sourceanalytix.0 (1082) Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
2021-08-10 00:00:00.026 - error: sourceanalytix.0 (1082) unhandled promise rejection: Cannot read property 'captureException' of undefined
2021-08-10 00:00:00.027 - error: sourceanalytix.0 (1082) TypeError: Cannot read property 'captureException' of undefined
at Sourceanalytix.errorHandling (/opt/iobroker/node_modules/iobroker.sourceanalytix/main.js:1647:37)
at CronJob. (/opt/iobroker/node_modules/iobroker.sourceanalytix/main.js:849:12)
2021-08-10 00:00:00.027 - error: sourceanalytix.0 (1082) Cannot read property 'captureException' of undefined
2021-08-10 00:00:00.031 - info: sourceanalytix.0 (1082) SourceAnalytix stopped, now you have to calculate by yourself :'( ...
2021-08-10 00:00:00.032 - info: sourceanalytix.0 (1082) terminating
2021-08-10 00:00:00.035 - warn: sourceanalytix.0 (1082) Terminated (UNCAUGHT_EXCEPTION): Without reason
2021-08-10 00:00:00.060 - info: host.buanet-iobroker1 instance system.adapter.daswetter.0 started with pid 317
```

Das wird verursacht durch shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned, dieser Datenpunkt ist **nicht** richtig konfiguriert was man auch beim starten von SA in den Fehlermeldungen sehen kan : 

"2021-08-10 00:00:32.076 - error: sourceanalytix.0 (360) No cost type defined for shelly.0.SHEM-3#8CAAB561FC70#1.Total.Total_Returned, please Select Type of calculation at state setting
"

Stelle den state mal bitte richtig ein, also einen kosten typen zuweisen, dan sollte es funktionieren.
Ich kümmere mich darum den bug vom adapter crash ab zu fangen

**Comment by DutchmanNL** [comment]
das lustige, es ist nicht mal der Fehler selber der den adapter crashen lässt sonder der error Handler :/, ich mache nen hoffix. :/

**Comment by serpentes80** [comment]
Ah 
schön doof von mir... :-/ vor allem: weil der State eh überflüssig war.
Hab ihn rausgenommen, nun kommen auch keine Fehler beim Restart der SA Instanz.
kann man den Reset der Tageswerte irgendwie manuell triggern, oder müssen wir bis morgen abwarten, obs dann funktioniert? :-)

Grüße

Marcel

**Comment by DutchmanNL** [comment]
> Ah
> schön doof von mir... :-/ vor allem: weil der State eh überflüssig war.
> Hab ihn rausgenommen, nun kommen auch keine Fehler beim Restart der SA Instanz.
> kann man den Reset der Tageswerte irgendwie manuell triggern, oder müssen wir bis morgen abwarten, obs dann funktioniert? :-)
> 
> Grüße
> 
> Marcel

da musst du bis morgen warten, wird durch den Zeitpunkt 00.00 ausgelöst :)

**Comment by serpentes80** [comment]
OK,
dann warten wir mal ab-, was heute Nacht passiert. Ich geb dann Bescheid 👍

**Comment by DutchmanNL** [comment]
bugfix released in 0.4.10,  das wird anderen auch freuen danke fürs melden und log details !
